src/test/clojure/clara/performance/test_compilation.clj

```diff
@@ -0,0 +1,154 @@
+                    production (assoc base-production
+                                 :lhs [{:accumulator '(clara.rules.accumulators/all)
+                                        :from {:type (keyword seed-sym),
+                                               :constraints [`(filter-fn ~'this)]}
+                                        :result-binding :?binding}]
+                                 :rhs `(r/insert! (with-meta ~'?binding
+                                                             {:type ~(keyword next-fact)})))]]
+          [next-fact production])))
+
+(defn generate-queries
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [production (assoc base-production
+                                 :lhs [{:type [seed-sym]
+                                        :constraints []
+                                        :fact-binding :?binding}]
+                                 :params #{})]]
+          [seed-sym production])))
+
+(defn generate-rules-and-opts
+  [num-starter-facts]
+  (let [starter-seeds (repeatedly num-starter-facts gensym)]
+    (-> (reduce (fn [[seeds prods] next-fn]
+                  (let [new-productions-and-facts (next-fn seeds)]
+                    [(keys new-productions-and-facts)
+                     (concat prods
+                             (vals new-productions-and-facts))]))
+                [starter-seeds []]
+                [generate-filter-productions
+                 generate-compose-productions
+                 generate-collection-productions
+                 generate-queries])
+        second
+        vector
+        (conj :cache false))))
+
+(deftest compilation-performance-test
+  (let [rules (generate-rules-and-opts 500)]
+    (testing "Session creation performance"
+      (run-performance-test
+        {:description "Generated Session Compilation"
+         :func #(com/mk-session rules)
+         :iterations 50
+         :mean-assertion (partial > 5000)}))
+
+    (let [session (com/mk-session rules)
+          os (ByteArrayOutputStream.)]
+      (testing "Session serialization performance"
+        (run-performance-test
+          {:description "Generated Session serialization"
+           :func #(.serialize (dura/create-session-serializer (ByteArrayOutputStream.))
+                              session
+                              {:rulebase-only? true})
+           :iterations 50
+           :mean-assertion (partial > 1000)}))
+
+      (testing "Session deserialization performance"
+        (.serialize (dura/create-session-serializer os)
+                    session
+                    {:rulebase-only? true})
+
+        (let [session-bytes (.toByteArray os)]
+          (run-performance-test
+            {:description "Generated Session deserialization"
+             :func #(-> (dura/create-session-serializer (ByteArrayInputStream. session-bytes))
+                        (.deserialize session-bytes {:rulebase-only? true}))
```

Comment:
Why is `:rulebase-only?` only used in the last deserialize case? Shouldn't it be set for all serialize and deserialize calls?